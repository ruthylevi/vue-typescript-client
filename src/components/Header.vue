<template>
  <v-toolbar dark height="100">
    <router-link to="/">
      <img class="logo" :src="require('@/../public/assets/logo.png')" />

      <img :src="require('@/../public/assets/title.png')" height="50px" />
    </router-link>
    <v-spacer></v-spacer>
    <v-toolbar-items>
      <v-btn text class="item" to="/">Projects</v-btn>
      <v-btn text class="item" to="/my-projects">My Projects</v-btn>
      <v-btn text class="item" to="/create">Create Project</v-btn>
      <v-btn text class="item" to="/applications">Applications</v-btn>
      <v-btn text class="item" to="/about">About</v-btn>
    </v-toolbar-items>
    <v-menu offset-y>
      <template v-slot:activator="{ on }">
        <v-btn icon v-on="on">
          <v-icon>mdi-bell</v-icon>
        </v-btn>
      </template>
      <v-list>
        <v-list-item
          v-for="(item, index) in items"
          :key="index"
          @click="router.push('/applications')"
        >
          <v-list-item-title>{{ item.title }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
    <v-btn icon to="/account">
      <v-icon>mdi-account-circle</v-icon>
    </v-btn>
    <v-btn
      icon
      href="https://marqetplace.auth.us-east-1.amazoncognito.com/oauth2/authorize?identity_provider=stevens-shibboleth&redirect_uri=https://www.marqetplace.xyz&response_type=TOKEN&client_id=6893005so6v9k2kuunc4acckps"
    >
      <v-icon>mdi-account-circle</v-icon>
    </v-btn>
    <v-btn icon @click="logout">
      Logout
    </v-btn>
  </v-toolbar>
</template>

<script>
import store from '@/store';

export default {
  data: () => ({
    items: [
      { title: 'Click Me' },
      { title: 'Click Me' },
      { title: 'Click Me' },
      { title: 'Click Me 2' },
    ],
  }),
  methods: {
    logout() {
      store.commit('resetToken');
    },
  },
};
</script>

<style>
.title {
  font-family: "Montserrat", sans-serif;
  font-size: 30px !important;
  font-weight: bold;
}
.item {
  font-family: "Montserrat", sans-serif;
  font-size: 15px !important;
  font-weight: bold !important;
}
.logo {
  width: 50px;
  height: 50px;
}
</style>
