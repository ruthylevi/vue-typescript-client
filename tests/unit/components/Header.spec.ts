import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import Header from '@/components/Header.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const wrapper = shallowMount(Header);

describe('Header', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
