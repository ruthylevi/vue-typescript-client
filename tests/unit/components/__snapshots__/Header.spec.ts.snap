// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders correctly 1`] = `
<v-toolbar-stub height="100" dark="true" tag="header" tile="true" extensionheight="48" src="">
  <router-link-stub to="/" tag="a" event="click"><img src="" class="logo"> <img src="" height="50px"></router-link-stub>
  <v-spacer-stub></v-spacer-stub>
  <v-toolbar-items-stub>
    <v-btn-stub tag="button" activeclass="" to="/" text="true" type="button" class="item">Projects</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" to="/my-projects" text="true" type="button" class="item">My Projects</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" to="/create" text="true" type="button" class="item">Create Project</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" to="/applications" text="true" type="button" class="item">Applications</v-btn-stub>
    <v-btn-stub tag="button" activeclass="" to="/about" text="true" type="button" class="item">About</v-btn-stub>
  </v-toolbar-items-stub>
  <v-menu-stub opendelay="0" closedelay="0" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" offsety="true" origin="top left" transition="v-menu-transition">
    <v-list-stub tag="div" tile="true">
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>Click Me</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>Click Me</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>Click Me</v-list-item-title-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" tag="div">
        <v-list-item-title-stub>Click Me 2</v-list-item-title-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-menu-stub>
  <v-btn-stub tag="button" activeclass="" to="/account" icon="true" type="button">
    <v-icon-stub>mdi-account-circle</v-icon-stub>
  </v-btn-stub>
  <v-btn-stub tag="button" activeclass="" href="https://marqetplace.auth.us-east-1.amazoncognito.com/oauth2/authorize?identity_provider=stevens-shibboleth&amp;redirect_uri=https://www.marqetplace.xyz&amp;response_type=TOKEN&amp;client_id=6893005so6v9k2kuunc4acckps" icon="true" type="button">
    <v-icon-stub>mdi-account-circle</v-icon-stub>
  </v-btn-stub>
  <v-btn-stub tag="button" activeclass="" icon="true" type="button">
    Logout
  </v-btn-stub>
</v-toolbar-stub>
`;
